distributed/comm/tcp.py

```diff
@@ -35,6 +38,7 @@
 
 
 MAX_BUFFER_SIZE = MEMORY_LIMIT / 2
+C_MAX_INT = 256 ** ctypes.sizeof(ctypes.c_int) // 2 - 1
```

Comment:
AFAICT the limit to what OpenSSL can handle is based on the maximum value of `int` in C (as [used by `num` bytes]( https://linux.die.net/man/3/ssl_read )). This is implementation specific. So am using `ctypes` to figure that out.

Reply:
Just to back this up, the following C program gives this result

```c
#include <stdio.h>
#include <limits.h>


int main() {
	printf("%lu\n", INT_MAX);
	return 0;
}
```

```
2147483647
```

Using this Python code, I get the same result.

```python
import ctypes

print(256 ** ctypes.sizeof(ctypes.c_int) // 2 - 1)
```

Reply:
Renamed the variable to `C_INT_MAX`. Otherwise the code is the same